File: src/com/sterndu/json/FloatValue.java
package com.sterndu.json;

import java.util.Locale;
import java.util.function.Function;

public class FloatValue implements NumberValue {

	private float value = 0.0f;

	public FloatValue(float value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public float getValueFloat() { return value; }

	public void setValue(float value) { this.value = value; }

	@Override
	public String toJson() {
		return String.format(Locale.US, "%g", value);
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return String.format(Locale.US, "%g", value);
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/FloatValue.kt
@file:JvmName("FloatValue")
package com.sterndu.json

import java.util.*

class FloatValue @JvmOverloads constructor(value: Float = 0.0f) : NumberValue {
	private var valueFloat = value

	fun getValueFloat(): Float {
		return valueFloat
	}

	override fun getValue(): Number {
		return valueFloat
	}

	fun setValue(value: Float) {
		valueFloat = value
	}

	override fun toJson(): String {
		return String.format(Locale.US, "%g", valueFloat)
	}

	override fun toJson(function: (Any?) -> String): String {
		return String.format(Locale.US, "%g", valueFloat)
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
